### PR TITLE
CI/CD for fossil Offchain Processor 

### PR DESCRIPTION
**Pull request by adityaNth**

### CODE REVIEW

File: .github/workflows/deploy_fossil_offchain.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          mask-password: "true"
+
+      - name: Check if Docker is installed
+        run: |
+          if ! command -v docker &> /dev/null
+          then
+            sudo apt-get update
+            sudo apt-get install -y docker.io
+            sudo apt-get install -y docker-compose
+          fi
```
Comment (cwkang1998) [inline .github/workflows/deploy_fossil_offchain.yml]: Pretty sure docker is already available in github actions by default, but if not I prefer to use the actions below:

```
      - name: Set up QEMU
        uses: docker/setup-qemu-action@v3
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3
```

This action will properly setup buildx depedencies and multi platform builds as well, so I think its a better alternative.

File: .github/workflows/deploy_fossil_offchain.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
+name: Build and Run Yokai Agent Docker Container
```
Comment (cwkang1998) [inline .github/workflows/deploy_fossil_offchain.yml]: remember to change the name!

File: .github/workflows/deploy_fossil_offchain.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          IMAGE_URI=$ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
+          echo "IMAGE_URI=$IMAGE_URI" >> $GITHUB_ENV
+          docker buildx build --platform=linux/amd64 -f ./crates/server/Dockerfile -t $IMAGE_URI .
+          docker push $IMAGE_URI
+
+      - name: Download task definition
+        run: |
+          aws ecs describe-task-definition --task-definition ${{ env.TASK_NAME }} --query taskDefinition > task-definition.json
+
+      # - name: Read secrets from AWS Secrets Manager into environment variables
```
Comment (cwkang1998) [inline .github/workflows/deploy_fossil_offchain.yml]: Let's remove commented information that we do not need here.

**Review by cwkang1998 — COMMENTED** [review]
Generally looks good, minor comments and requests for changes.

File: .github/workflows/deploy_fossil_offchain.yml (hunk 3 above)
Comment (adityaNth) [reply .github/workflows/deploy_fossil_offchain.yml]: done

**Review by cwkang1998 — APPROVED** [review]

**Pull request merged**